projects/plugins/boost/app/features/optimizations/cloud-css/Cloud_CSS_Request.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Cloud CSS state.
+ *
+ * @link       https://automattic.com
+ * @since      1.0.0
+ * @package    automattic/jetpack-boost
+ */
+
+namespace Automattic\Jetpack_Boost\Features\Optimizations\Cloud_CSS;
+
+use Automattic\Jetpack_Boost\Lib\Boost_API;
+use Automattic\Jetpack_Boost\Lib\Critical_CSS\Critical_CSS_State;
+use Automattic\Jetpack_Boost\Lib\Critical_CSS\Source_Providers\Source_Providers;
+
+/**
+ * Cloud CSS State
+ */
+class Cloud_CSS_Request {
+
+	/**
+	 * @var Critical_CSS_State
+	 */
+	private $cloud_css_state;
+
+	/**
+	 * @var Source_Providers
+	 */
+	private $source_providers;
+
+	public function __construct() {
+		$this->source_providers = new Source_Providers();
+		$this->cloud_css_state  = new Critical_CSS_State();
```

Comment:
Is it safe to create multiple of these as needed? Each one will read and write (and store) a copy of the transient separately.

Wouldn't it be safer to pass in a Critical_CSS_State object?

Reply:
I have refactored the `Critical_CSS_State` a bit. It is now possible to instantiate different states by naming them differently. You can also pass an instance to the request class.